qemu/tests/qemu_guest_agent.py

```diff
@@ -390,6 +405,260 @@ def gagent_check_set_vcpus(self, test, params, env):
+        """
+        timeout = float(params.get("login_timeout", 240))
+        session = self.vm.wait_for_login(timeout=timeout)
+        get_guest_time_cmd = params["get_guest_time_cmd"]
+        error.context("get the time of the guest", logging.info)
+        guest_time_before = session.cmd_output(get_guest_time_cmd)
+        if not guest_time_before:
+            raise error.TestError("can't get the guest time for contrast")
+        error.context("the time before being moved back into past  is '%d' "
+                      % int(guest_time_before), logging.info)
+        # Need to move the guest time one week into the past
+        target_time = (int(guest_time_before) - 604800) * 1000000000
+        self.gagent.set_time(target_time)
+        guest_time_after = session.cmd_output(get_guest_time_cmd)
+        error.context("the time after being moved back into past  is '%d' "
+                      % int(guest_time_after), logging.info)
+        delta = abs(int(guest_time_after) - target_time / 1000000000)
+        if delta > 3:
+            raise error.TestFail("the time set for guest is not the same "
+                                 "with target")
+        # Set the system time from the hwclock
+        if params["os_type"] != "windows":
+            move_time_cmd = params["move_time_cmd"]
+            session.cmd("hwclock -w")
+            guest_hwclock_after_set = session.cmd_output("date +%s")
+            error.context("hwclock is '%d' " % int(guest_hwclock_after_set),
+                          logging.info)
+            session.cmd(move_time_cmd)
+            time_after_move = session.cmd_output("date +%s")
+            error.context("the time after move back is '%d' "
+                          % int(time_after_move), logging.info)
+            self.gagent.set_time()
+            guest_time_after_reset = session.cmd_output(get_guest_time_cmd)
+            error.context("the time after being reset is '%d' "
+                          % int(guest_time_after_reset), logging.info)
+            guest_hwclock = session.cmd_output("date +%s")
+            error.context("hwclock for compare is '%d' " % int(guest_hwclock),
+                          logging.info)
+            delta = abs(int(guest_time_after_reset) - int(guest_hwclock))
+            if delta > 3:
+                raise error.TestFail("The guest time can't be set from hwclock"
+                                     " on host")
+
+    @error.context_aware
+    def gagent_check_fstrim(self, test, params, env):
+        """
+        Execute "guest-fstrim" command to guest agent
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+
+        """
+        def get_host_scsi_disk():
+            """
+            Get latest scsi disk which enulated by scsi_debug module
+            Return the device name and the id in host
+            """
+            os_dep.command("lsscsi")
```

Comment:
Avocado has it owned utils to look up a command on host.
```python
>>> from avocado.utils import path as avo_path
>>> avo_path.find_command('lsscsi')
'/bin/lsscsi'
```
so you can change the following line like
```python
scsi_disk_info = process.system_output(
    avo_path.find_command('lsscsi'),
    shell=True
).splitlines()
```